Assets/Scripts/Player/PlayerController.cs

```diff
@@ -3,22 +3,39 @@
 using System.Collections;
 using System.Collections.Generic;
 using Bullet;
+using Inputs;
+using UI;
 
 namespace Player
 {
+    public class PlayerData
+    {
+        public int score, health;
+    }
+
     public class PlayerController
     {
         public PlayerModel playerModel { get; private set; }
         public PlayerView playerView { get; private set; }
+        public InputComponent playerInput { get; private set; }
 
-        public PlayerController()
+        public PlayerData playerData { get; private set; }
+
+        private float lastTime;
+
+        public PlayerController(InputComponentScriptable inputComponentScriptable)
         {
             GameObject prefab = Resources.Load<GameObject>("Tank");
             GameObject tankObj = GameObject.Instantiate<GameObject>(prefab);
 
             playerModel = new PlayerModel();
+            playerInput = new InputComponent();
+            playerInput.SetController(this);
```

Comment:
since setting controller is mandatory - move it to the constructor argument instead